stl/inc/iterator

```diff
@@ -25,12 +25,18 @@ class back_insert_iterator { // wrap pushes to back of container as output itera
 public:
     using iterator_category = output_iterator_tag;
     using value_type        = void;
-    using difference_type   = void;
     using pointer           = void;
     using reference         = void;
 
     using container_type = _Container;
 
+#ifdef __cpp_lib_concepts
+    using difference_type                     = ptrdiff_t;

```

Comment:
I would like to see a newline since there is no commonality between the typedef and the defaulted constructor. (Even if we didn't have align consecutive assignments). Ditto below.